### PR TITLE
Np 9220 implement email sending

### PR DESCRIPTION
**Pull request by anetteOlli**

### CODE REVIEW

File: email-service/src/test/java/no/sikt/nva/email/EmailRequestHandlerTest.java [hunk 1]
```diff
@@ -52,19 +53,22 @@ public void init() {
...
                                              "test2@test.no",
                                              "test3.test.no",
-                                             "test4@test.no",
                                              randomString(),
                                              randomString(),
                                              randomString());
     }
 
     @Test
     public void sendsEmailSuccessfully() throws ApiGatewayException {
```
Comment (axthosarouris) [inline email-service/src/test/java/no/sikt/nva/email/EmailRequestHandlerTest.java:62]: when does it send email successfully?

File: email-service/src/main/java/no/sikt/nva/email/EmailRequestHandler.java [hunk 2]
```diff
@@ -35,14 +46,68 @@ public EmailRequestHandler() {
...
+    }
+
+    private Destination getDestination(EmailRequest emailRequest) {
+        return new Destination()
+                   .withToAddresses(emailRequest.getToName())
+                   .withCcAddresses(emailRequest.getCc())
+                   .withBccAddresses(emailRequest.getBcc());
+    }
+
+    @SuppressWarnings("PMD.InvalidLogMessageFormat")
```
Comment (axthosarouris) [inline email-service/src/main/java/no/sikt/nva/email/EmailRequestHandler.java:104]: why are you suppressing the warning

File: email-service/src/main/java/no/sikt/nva/email/EmailRequestHandler.java [hunk 3]
```diff
@@ -35,14 +46,68 @@ public EmailRequestHandler() {
...
         return HttpURLConnection.HTTP_OK;
     }
+
+    private SendEmailResult sendEmail(EmailRequest emailRequest) throws EmailException {
+        return attempt(() -> createSesEmailRequest(emailRequest))
+                   .map(amazonSimpleEmailService::sendEmail)
+                   .orElseThrow(this::logFailureAndThrowEmailException);
+    }
+
+    @SuppressWarnings("PMD.InvalidLogMessageFormat")
```
Comment (axthosarouris) [inline email-service/src/main/java/no/sikt/nva/email/EmailRequestHandler.java:68]: update the PMD config instead of suppressing the warning all the time

**Review by axthosarouris — CHANGES_REQUESTED** [review]

File: email-service/src/main/java/no/sikt/nva/email/EmailRequestHandler.java (hunk 2 above)
Comment (anetteOlli) [reply email-service/src/main/java/no/sikt/nva/email/EmailRequestHandler.java:104]: codacy was complaining. There's no problem with the method locally. It's complaining that I have too many arguments. "Expected 0 arguments got 1"

File: email-service/src/main/java/no/sikt/nva/email/EmailRequestHandler.java (hunk 3 above)
Comment (anetteOlli) [reply email-service/src/main/java/no/sikt/nva/email/EmailRequestHandler.java:68]: codacy bug is the reason, not that I don't like to get warnings if the logMessage format is incorrect.

File: email-service/src/test/java/no/sikt/nva/email/EmailRequestHandlerTest.java (hunk 1 above)
Comment (anetteOlli) [reply email-service/src/test/java/no/sikt/nva/email/EmailRequestHandlerTest.java]: Changed name of method.

I think the test if it possible to send mail should perhaps be an integration test? 
Or is it slightly overkill?
I set up APIgateway to check the input body. (and tested it).

File: README.md [hunk 4]
```diff
@@ -1 +1,26 @@
...
+## Sending requests
+from_name, to_name and subject are required fields.
+Atleast one of text and text_html must be present.
+
+### Sample
+POST:
+```json
+    {
+  "from_name": "noreply@test.com",
+  "to_name": "test1@test.com",
```
Comment (nraanes) [inline README.md:19]: Kanskje bedre om det heter "to_email", "to_address" eller "to_email_address"? Samme tenker jeg på "from_name"

Comment (anetteOlli) [reply README.md]: Fikset

**Pull request merged**